File: 386. Lexicographical Numbers/lexicalOrder.py
from typing import List

class Solution:
  def lexicalOrder(self, n: int) -> List[int]:
    lecsicographicalOrder = []
    current = 1

    for i in range(n):
      lecsicographicalOrder.append(current)

      if current * 10 <= n:
        current *= 10
      else:
        if current >= n:
          current //= 10
        current += 1
        print('current: ', current)
        while current % 10 == 0:
          current //= 10

    return lecsicographicalOrder
  

n = 131
solution = Solution()
print('lexicalOrder: ', solution.lexicalOrder(n)) # [1, 10, 11, 12, 13, 2, 3, 4, 5, 6, 7, 8, 9]

File: 386. Lexicographical Numbers/lexicalOrder.js
/**
 * @param {number} n
 * @return {number[]}
 */
const lexicalOrder = (n) => {
  const lecsicographicalOrder = [];
  let current = 1;

  for (let i = 1; i <= n; i++) {
    lecsicographicalOrder.push(current);
    if (current * 10 <= n) {
      current *= 10;
    } else {
      if (current >= n) {
        current = Math.floor(current / 10);
      }
      current++;
      while (current % 10 === 0) {
        current = Math.floor(current / 10);
      }
    }
  }

  return lecsicographicalOrder;
};


const n = 13;
console.log('lexicalOrder: ', lexicalOrder(n)); // [1,10,11,12,13,2,3,4,5,6,7,8,9]
